include/seqan3/range/views/minimiser.hpp

```diff
@@ -0,0 +1,481 @@
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \author Mitra Darvish <mitra.darvish AT fu-berlin.de>
+ * \brief Provides seqan3::views::minimiser.
+ */
+
+#pragma once
+
+#include <deque>
+
+#include <seqan3/range/concept.hpp>
+#include <seqan3/range/views/detail.hpp>
+
+namespace seqan3::detail
+{
+// ---------------------------------------------------------------------------------------------------------------------
+// minimiser_view class
+// ---------------------------------------------------------------------------------------------------------------------
+
+/*!\brief The type returned by seqan3::views::minimiser.
+ * \tparam urng_t The type of the underlying ranges, must model std::forward_range, the reference type must model
+ *                std::totally_ordered. The typical use case is that the reference type is the result of
+ *                seqan3::kmer_hash.
+ * \implements std::ranges::view
+ * \implements std::ranges::random_access_range
+ * \implements std::ranges::sized_range
+ * \ingroup views
+ *
+ * \details
+ *
+ * Note that most members of this class are generated by ranges::view_interface which is not yet documented here.
+ */
+template <std::ranges::view urng_t>
+class minimiser_view : public std::ranges::view_interface<minimiser_view<urng_t>>
+{
+private:
+    static_assert(std::ranges::forward_range<urng_t const>, "The minimiser_view only works on forward_ranges.");
+    static_assert(std::totally_ordered<reference_t<urng_t>>, "The reference type of the underlying range must model "
+                                                             "std::totally_ordered.");
+    //!\brief The underlying range.
+    urng_t urange;
+
+    //!\brief The number of values in one window.
+    uint32_t window_values_size;
+
+    template <typename rng_t>
+    class window_iterator;
+
+public:
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    minimiser_view()                                       = default; //!< Defaulted.
+    minimiser_view(minimiser_view const & rhs)             = default; //!< Defaulted.
+    minimiser_view(minimiser_view && rhs)                  = default; //!< Defaulted.
+    minimiser_view & operator=(minimiser_view const & rhs) = default; //!< Defaulted.
+    minimiser_view & operator=(minimiser_view && rhs)      = default; //!< Defaulted.
+    ~minimiser_view()                                      = default; //!< Defaulted.
+
+    //!\brief Construct from a view and a given number of values in one window.
+    minimiser_view(urng_t urange_, uint32_t const & w_) : urange{std::move(urange_)}, window_values_size{w_}{}
```

Comment:
```suggestion
    minimiser_view(urng_t urange_, uint32_t const w) : urange{std::move(urange_)}, window_values_size{w_}{}
```
And the documentation `\param` is missing
Same for the contructor below